core/src/main/java/org/infinispan/topology/LocalTopologyManagerImpl.java

```diff
@@ -82,20 +91,26 @@ public void stop() {
          log.tracef("Stopping LocalTopologyManager on %s", transport.getAddress());
       }
       running = false;
+      withinThreadExecutor.shutdown();
```

Comment:
I thought calling shutdown would prevent other tasks from running, but I guess not... I'll leave it there in case we do change the implementation at some point.